doc/user/content/sql/alter-sink.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "ALTER SINK"
+description: "`ALTER SINK` changes certain characteristics of a sink."
+menu:
+  main:
+    parent: 'commands'
+---
+
+`ALTER SINK` changes certain characteristics of a sink.
+
+## Syntax
+
+{{< diagram "alter-sink.svg" >}}
+
+## Context
+
+Altering the `FROM` item of a sink results in Materialize picking a consistent
+cutover timestamp. The resulting topic will contain all the updates of the old
+item that happened before the cutover timestamp and all the updates of the new
+item that happened after the cutover timestamp. Care must be taken to ensure
+that the old and the new items make sense when stitched together. See the
+anomalies section for more details on what can be observed.
+
+**Note**: In order to select a consistent timestamp Materialize waits for the
+previous definition of the sink to make enough progress such that there is
+overlap between the available timestamps of the desired new item. Attempting to
+`ALTER` an unhealthy sink that can't make progress will result to the command
+timing out.
+
+### Valid schema changes
+
+Materialize allows altering a sink to a new item provided that the new item
+leads to a valid sink definition with the same rules as the `CREATE SINK`
+statement.
+
+When using the Avro format with a schema registry care must be taken such that
+the generated Avro schema of the new item is compatible with the previously
+published schema. When that is not the case the `ALTER SINK` statement will
+succeed but the subsequent execution of the sink will result in errors and will
+be unable to make progress.
+
+### Possible anomalies
+
+Depending on the contents and state of the old and new items a few different
+anomalies can be observed in the output topic as a result of altering a sink.
+These anomalies occur because `ALTER SINK` causes Materialize to emit updates
+from the new item only if they happen after the cutover timestamp.
+
+**Anomaly 1: Topic contains stale value for a key**
+
+As a consequence of not re-emitting a snapshot after `ALTER SINK`, all keys of
+the collection will appear to have the value they had in the old collection
+until an update happens to them. At that point the current value will be
+published to the topic.
+
+It is important that readers of the topic are prepared to handle an old value
+for a key, for example by filling in additional columns with default values.
+Alternatively, forcing an update to all the keys after `ALTER SINK` will force
```

Comment:
Does it make sense to mention that dropping and recreating the sink emits a complete snapshot to guild folks who should NOT use `ALTER SINK in` the right direction?

Reply:
Is there a use case for `ALTER SINK` outside blue/green deployments right now? I'd like to frame this in the clearest possible way. If there isn't, no one should be manually running this command, and these docs should point very early on to the dbt workflow. I'm pushing some updates soon.